From fb50a17bf242ffe88be58f83d710277e22ad2279 Mon Sep 17 00:00:00 2001
From: Stanislaw Kardach <kda@semihalf.com>
Date: Mon, 7 Aug 2017 16:09:11 +0200
Subject: [PATCH 216/375] octeontx: only require PKI VF if domain has ports

Change domain requirements so that it does not require PKI VF if there
are no ports specified in the Application Domain specification.

Fix build issues.

Signed-off-by: Stanislaw Kardach <kda@semihalf.com>
Signed-off-by: Chandrakala Chavva <cchavva@caviumnetworks.com>
---
 drivers/net/ethernet/cavium/octeontx-83xx/octeontx_main.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/cavium/octeontx-83xx/octeontx_main.c b/drivers/net/ethernet/cavium/octeontx-83xx/octeontx_main.c
index b24da1bc60f1..6f1ba62e3361 100644
--- a/drivers/net/ethernet/cavium/octeontx-83xx/octeontx_main.c
+++ b/drivers/net/ethernet/cavium/octeontx-83xx/octeontx_main.c
@@ -521,8 +521,8 @@ int octeontx_create_domain(const char *name, int type, int sso_count,
 		return -EINVAL;
 	}
 
-	if (!pki_count) {
-		dev_err(octtx_device, "Domain has to include exactly 1 PKI\n");
+	if ((bgx_count + lbk_count) != 0 && pki_count != 1) {
+		dev_err(octtx_device, "Domain has to include exactly 1 PKI if there are BGX or LBK ports\n");
 		return -EINVAL;
 	}
 
-- 
2.14.1

